Look for commercial PyQt wheel to determine layout

lv-develer opened the pull request:
Commercial PyQt wheels install with the "_commercial" suffix as their package name but still make the regular PyQt5/PyQt6 namespace packages available.

This leads the layout detection logic to assume the old layout because `is_module_satisfies()` enters a fallback code path where it looks for the `__version__` attribute inside the namespace package, which doesn't exist.

A possible solution is to look for package names with the "_commercial" suffix when the standard lookup fails. The `_use_new_layout` utility method was introduced to perform such checks and is then used in both PyQt5 and PyQt6 code paths.

rokm commented on PyInstaller/utils/hooks/qt/__init__.py:
```diff
@@ -181,6 +160,30 @@ def __getattr__(self, name):
 
+    # Check whether we can use the new layout (e.g. PyQt5/Qt5, PyQt6/Qt6) instead of the old layout (PyQt5/Qt, PyQt6/Qt)
+    @staticmethod
+    def _can_use_new_layout(requirements: str, requirements_commercial: str, fallback: bool) -> bool:
```
Two nitpicks here:

1. I would prefer if this was named just `_use_new_layout` (without can, because that implies an option, which is not really there). Same in the comment ("Check whether we *can* use...") - we actually must use the new layout.

2. For the arguments, I would prefer to see `package_basename`, `version`, and `fallback_value` (or `default_value`).

The helper would then be invoked like `self._use_new_layout("PyQt5", "5.15.4", False)`, and it would constructs both strings for `is_module_satisfies` on its own.

Regarding `fallback_value` vs just `fallback`, the latter implies to me that this is a switch that will enabled a fallback within the helper, as opposed to the fallback/default value returned by the helper.

lv-develer replied:
I should have addressed these points. Let me know if there's something else to do.

rokm commented:
LGTM. 

Maybe add a [news fragment](https://github.com/pyinstaller/pyinstaller/blob/develop/doc/development/changelog-entries.rst) for the changelog (i.e., create `news/7770.hooks.rst` with something along the lines of `Add support for commercial PyQt5 and PyQt6 wheels.`), and then we should be good to merge.

lv-develer committed:
hookutils: qt: look for commercial pyqt to determine layout

Commercial PyQt wheels install with the "_commercial" suffix as their
package name but still make the regular PyQt5/PyQt6 namespace packages
available.

This leads the layout detection logic to assume the old layout because
is_module_satisfies() enters a fallback code path where it looks for the
__version__ attribute inside the namespace package, which doesn't exist.

A possible solution is to look for package names with the "_commercial"
suffix when the standard lookup fails. The _use_new_layout utility
method was introduced to perform such checks and is then used in both
PyQt5 and PyQt6 code paths.
